homeassistant/components/media_player/template_media_player.py

```diff
@@ -0,0 +1,194 @@
+
+    def __init__(self, name, config, hass):
+        """Initialize the device."""
+        self._name = name
+        self._turn_on_action = Script(hass, config.get(
+            CONF_ON_ACTION)) if config.get(CONF_ON_ACTION) else None
+        self._turn_off_action = Script(hass, config.get(
+            CONF_OFF_ACTION)) if config.get(CONF_OFF_ACTION) else None
+        self._volume_up_action = Script(hass, config.get(
+            CONF_VOLUME_UP_ACTION)) if config.get(
+            CONF_VOLUME_UP_ACTION) else None
+        self._volume_down_action = Script(hass, config.get(
+            CONF_VOLUME_DOWN_ACTION)) if config.get(
+            CONF_VOLUME_DOWN_ACTION) else None
+        self._volume_set_action = Script(hass, config.get(
+            CONF_VOLUME_SET_ACTION)) if config.get(
+            CONF_VOLUME_SET_ACTION) else None
+        self._volume_mute_action = Script(hass, config.get(
+            CONF_VOLUME_MUTE_ACTION)) if config.get(
+            CONF_VOLUME_MUTE_ACTION) else None
+        self._volume_unmute_action = Script(hass, config.get(
+            CONF_VOLUME_UNMUTE_ACTION)) if config.get(
+            CONF_VOLUME_UNMUTE_ACTION) else None
+        self._select_source_action = Script(hass, config.get(
+            CONF_SELECT_SOURCE_ACTION)) if config.get(
+            CONF_SELECT_SOURCE_ACTION) else None
+        self._sources = config.get(CONF_SOURCE_NAMES)
+        self._selected_source = None
+        self._state = STATE_OFF
+        self._volume = config.get(CONF_INITIAL_VOLUME)
+        self._volume_step = config.get(CONF_VOLUME_STEP)
+        self._volume_step_delay = config.get(CONF_VOLUME_STEP_DELAY)
+        self._is_muted = False
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def supported_features(self):
+        """Flag media player features that are supported."""
+        features = 0
+        if self._turn_on_action:
+            features |= SUPPORT_TURN_ON
+        if self._turn_off_action:
+            features |= SUPPORT_TURN_OFF
+        if (self._volume_up_action and self._volume_down_action) or \
+                self._volume_set_action:
+            features |= SUPPORT_VOLUME_SET
+        if self._volume_mute_action and self._volume_unmute_action:
+            features |= SUPPORT_VOLUME_MUTE
+        if self._sources and self._select_source_action:
+            features |= SUPPORT_SELECT_SOURCE
+        return features
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        return list(self._sources.values())
+
+    def turn_on(self):
+        """Turn on media player."""
+        self._state = STATE_ON
+        if self._turn_on_action:
+            self._turn_on_action.run()
```

Comment:
All these methods need to be rewritten to be async and then do `yield from self._turn_on_action.async_run()`